athenahealth/lab_results.go

```diff
@@ -0,0 +1,303 @@
+	NoteToPatient      *string                 `json:"notetopatient"`
+	ObservationDate    *time.Time              `json:"observationdate"`
+	OriginalFilename   *string                 `json:"originalfilename"`
+	// 1 = high, 2 = normal
+	Priority    *string `json:"priority"`
+	ResultNotes *string `json:"resultnotes"`
+	// Final, Partial, Pending, Preliminary, Corrected, Cancelled
+	ResultStatus *string `json:"resultstatus"`
+	TieToOrderID *int    `json:"tietoorderid"`
+}
+
+type addLabResultDocumentResponse struct {
+	ErrorMessage string `json:"errormessage"`
+	LabResultID  int    `json:"labresultid"`
+	Success      bool   `json:"success"`
+}
+
+// AddLabResultDocument creates a lab result document record of a specific patient
+//
+// POST /v1/{practiceid}/patients/{patientid}/documents/labresult
+//
+// https://docs.athenahealth.com/api/api-ref/document-type-lab-result#Add-lab-result-document-to-patient's-chart
+func (h *HTTPClient) AddLabResultDocumentReader(ctx context.Context, patientID string, departmentID string, opts *AddLabResultDocumentOptions) (int, error) {
+	form := NewFormURLEncoder()
+
+	form.AddString("departmentid", departmentID)
+
+	if opts != nil {
+		if opts.AttachmentContents != nil {
+			form.AddReader("attachmentcontents", newBase64Reader(opts.AttachmentContents))
+		}
+		if len(opts.AttachmentType) > 0 {
+			form.AddString("attachmenttype", string(opts.AttachmentType))
+		}
+		if opts.InternalNote != nil {
+			form.AddString("internalnote", string(*opts.InternalNote))
+		}
+		if opts.NoteToPatient != nil {
+			form.AddString("notetopatient", string(*opts.NoteToPatient))
+		}
+		if opts.ObservationDate != nil {
+			form.AddString("observationdate", opts.ObservationDate.Format("01/02/2006"))
+		}
+		if opts.OriginalFilename != nil {
+			form.AddString("originalfilename", string(*opts.OriginalFilename))
+		}
+		if opts.Priority != nil {
+			form.AddString("priority", string(*opts.Priority))
+		}
+		if opts.ResultNotes != nil {
+			form.AddString("resultnotes", string(*opts.ResultNotes))
+		}
+		if opts.ResultStatus != nil {
+			form.AddString("resultstatus", string(*opts.ResultStatus))
+		}
+		if opts.TieToOrderID != nil {
+			form.AddInt("tietoorderid", *opts.TieToOrderID)
+		}
+	}
+
+	out := &addLabResultDocumentResponse{}
+
+	_, err := h.PostFormReader(ctx, fmt.Sprintf("patients/%s/documents/labresult", patientID), form, out)
+	if err != nil {
+		return 0, err
+	}
+
+	if !out.Success {
+		return 0, errors.New(out.ErrorMessage)
+	}
+
+	return out.LabResultID, nil
+}
+
+type ListChangedLabResultsOptions struct {
+	ShowPortalOnly             *bool     `json:"showportalonly"`
+	LeaveUnprocessed           *bool     `json:"leaveunprocessed"`
+	ShowProcessedEndDateTime   time.Time `json:"showprocessedenddatetime"`
+	ShowProcessedStartDateTime time.Time `json:"showprocessedstartdatetime"`
+
+	Pagination *PaginationOptions
+}
+
+type ListChangedLabResultsChangedLabResult struct {
```

Comment:
thoughts on just making this `ChangedLabResult`?

Reply:
How could you not love the name `ListChangedLabResultsChangedLabResult`? 😁 jk good call 👍